### ISSUE TITLE
remove querystring parameters that have been used in template Bug?

### ISSUE THREAD

**Issue by wtf-boy** [ThreeMammals/Ocelot#948]
## Expected Behavior / New Feature
```json
"Routes":
   {
      "DownstreamPathTemplate": "/{url}",
      "DownstreamScheme": "http",
      "DownstreamHostAndPorts": [
        {
          "Host": "localhost",
          "Port": 5001
        }
      ],
      "LoadBalancerOptions": {
        "Type": "LeastConnection"
      },
      "UpstreamPathTemplate": "/form/{url}",
      "UpstreamHttpMethod": [
        "Get",
        "Post"
      ]
    }
```
RequestPath:

http://localhost:5000/form/3235?code=161_3235_3c6aada4&notifyurl=http%3a%2f%2fwww.afuyy.com

Since my UpstreamPathTemplate PlaceholderName happens to be {url}, and the request address contains the "notifyurl" parameter, and the "code" parameter contains 3235(which is actually formId), the result is forwarded downstream to the address becomes:

http://localhost:5001/3235?code=161_3235_3c6aada4&notifurl=http%3a%2f%2fwww.afuyy.com

notifyurl=>notifurl  (Less a y)

reason:
context.DownstreamRequest.Query = context.DownstreamRequest.Query.Remove(questionMarkOrAmpersand - 1, 1);

on DownstreamUrlCreatorMiddleware.RemoveQueryStringParametersThatHaveBeenUsedInTemplate method
![image](https://user-images.githubusercontent.com/11848141/60331793-09e4a880-99c8-11e9-9048-0c08d06789a4.png)

## Specifications

  - Version:13.5.2
  - Platform:Linux version 3.10.0-693.2.2.el7.x86_64 (builder@kbuilder.dev.centos.org) (gcc version 4.8.5 20150623 (Red Hat 4.8.5-16) (GCC) ) 
  - Subsystem:

**Comment by wast** [comment]
Wow this is unbelievable. Can you still reproduce it?

**Comment by wtf-boy** [comment]
The placeholders in my UpstreamPathTemplate now try to avoid using the characters that appear in Query

> Wow this is unbelievable. Can you still reproduce it?

**Comment by wast** [comment]
Use {everything} and it will work.

**Comment by raman-m** [comment]
Current code:
https://github.com/ThreeMammals/Ocelot/blob/f4803c24bf9e9ca3929c78ca8eb23401e3c31c23/src/Ocelot/DownstreamUrlCreator/Middleware/DownstreamUrlCreatorMiddleware.cs#L120-L140

**Comment by raman-m** [comment]
This bug was fixed in commit https://github.com/ThreeMammals/Ocelot/commit/ec85b132a06b239faa499777820a9cc0143bb1d0
Fixed by PR #1335
